components/support/utils/src/test/java/mozilla/components/support/utils/SafeIntentTest.kt

```diff
@@ -4,23 +4,284 @@
+import android.os.Parcelable
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
 import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.anyBoolean
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.doReturn
 import org.mockito.Mockito.doThrow
 import org.mockito.Mockito.mock
 import org.robolectric.RobolectricTestRunner
+import java.util.HashSet
+import kotlin.collections.ArrayList
 
 @RunWith(RobolectricTestRunner::class)
 class SafeIntentTest {
+
+    private lateinit var intent: Intent
+
+    @Before
+    fun setup() {
+        intent = mock(Intent::class.java)
+    }
+
     @Test
     fun `getStringArrayListExtra returns null if intent throws OutOfMemoryError`() {
-        val intent = mock(Intent::class.java)
+
         doThrow(OutOfMemoryError::class.java)
-                .`when`(intent).getStringArrayListExtra(ArgumentMatchers.anyString())
+                .`when`(intent).getStringArrayListExtra(anyString())
 
         assertNull(SafeIntent(intent).getStringArrayListExtra("mozilla"))
     }
+
+    @Test
+    fun `getExtras returns null if intent throws OutOfMemoryError`() {
+        
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).extras
+
+        assertNull(SafeIntent(intent).extras)
+    }
+
+
+    @Test
+    fun `getAction return original action`() {
+
+        val expected = Intent.ACTION_MAIN
+
+        doReturn(expected)
+                .`when`(intent).action
+
+        assertEquals(expected, SafeIntent(intent).action)
+    }
+
+
+    @Test
+    fun `getFlags returns original flags`() {
+
+        val expected = Intent.FLAG_ACTIVITY_NEW_TASK
+
+        doReturn(expected)
+                .`when`(intent).flags
+
+        assertEquals(expected, SafeIntent(intent).flags)
+    }
+
+
+    @Test
+    fun `isLauncherIntent returns false if intent is not Launcher Intent`() {
+
+        // category is null
+        doReturn(null)
+                .`when`(intent).categories
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+        //both category and action are not valid
+        val category = HashSet<String>()
+
+        category.add("NOT" + Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn("NOT" + Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+        // action is not valid
+        category.clear()
+
+        category.add(Intent.CATEGORY_LAUNCHER)
+
+        doReturn(category)
+                .`when`(intent).categories
+
+        doReturn("NOT" + Intent.ACTION_MAIN)
+                .`when`(intent).action
+
+        assertFalse(SafeIntent(intent).isLauncherIntent)
+
+
```

Comment:
\[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1